compiler/rustc_monomorphize/src/collector/abi_check.rs

```diff
@@ -0,0 +1,111 @@
+use rustc_abi::Abi;
+use rustc_hir::CRATE_HIR_ID;
+use rustc_middle::ty::{self, Instance, InstanceKind, ParamEnv, Ty, TyCtxt};
+use rustc_session::lint::builtin::ABI_ERROR_DISABLED_VECTOR_TYPE;
+use rustc_span::def_id::DefId;
+use rustc_span::{Span, Symbol};
+use rustc_target::abi::call::{FnAbi, PassMode};
+
+use crate::errors::{AbiErrorDisabledVectorTypeCall, AbiErrorDisabledVectorTypeDef};
+
+// Represents the least-constraining feature that is required for vector types up to a certain size
+// to have their "proper" ABI.
+const X86_VECTOR_FEATURES: &'static [(u64, &'static str)] =
```

Comment:
This table should be moved to `compiler/rustc_target/src/target_features.rs` where all the other target feature info lives.

Please make sure the name of the constant includes "ABI".